### PR TITLE
perf(streaming): consume profiles in batches to improve write efficiency

### PR DESCRIPTION
**Pull request by chengjoey**
#### What this PR does / why we need it:
consume profiles in batches to improve write efficiency

#### Which issue(s) this PR fixes:

- Fixes #your-issue_number
- [Erda Cloud Issue Link](https://erda.cloud/erda/dop/projects/387/issues/all?id=445622&iterationID=1841&tab=TASK&type=TASK)

#### Specified Reviewers:

/assign @sfwn 

#### ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English | consume profiles in batches to improve write efficiency             |
| 🇨🇳 中文    |      优化了批量写入profiles数据提高写入的效率        |

#### Need cherry-pick to release versions?

Add comment like `/cherry-pick release/1.0` when this PR is merged.

> For details on the cherry pick process, see the [cherry pick requests](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md#how-to-cherry-pick-a-merged-pr) section under [CONTRIBUTING.md](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: cmd/monitor/streaming/bootstrap.yaml [hunk 1]
```diff
@@ -685,7 +689,13 @@ erda.oap.collector.exporter.clickhouse@metric:
     data_type: metric
     tenant_id_key: _metric_scope_id
 
-erda.oap.collector.exporter.pyroscope@profile: {}
+erda.oap.collector.exporter.clickhouse@profile:
+  _enable: ${WRITE_METRIC_TO_CLICKHOUSE_ENABLE:true}
```
Comment (sfwn) [inline cmd/monitor/streaming/bootstrap.yaml:693]: adjust env name

File: cmd/monitor/streaming/bootstrap.yaml [hunk 2]
```diff
@@ -685,7 +689,13 @@ erda.oap.collector.exporter.clickhouse@metric:
...
+  storage:
+    currency_num: ${EXPORTER_CH_METRIC_CURRENCY_NUM:2}
```
Comment (sfwn) [inline cmd/monitor/streaming/bootstrap.yaml:695]: same

File: cmd/monitor/streaming/bootstrap.yaml [hunk 3]
```diff
@@ -685,7 +689,13 @@ erda.oap.collector.exporter.clickhouse@metric:
...
+  builder:
+    data_type: profile
+    tenant_id_key: _metric_scope_id
```
Comment (sfwn) [inline cmd/monitor/streaming/bootstrap.yaml:698]: what's this

File: internal/tools/monitor/core/profile/profile.go [hunk 4]
```diff
@@ -67,6 +71,73 @@ func (p *ProfileIngest) Copy() *ProfileIngest {
...
+}
+
+func (o *Output) GetTags() map[string]string {
+	if len(o.profiles) == 0 {
+		return map[string]string{}
+	}
+	for _, v := range o.profiles {
+		if v.App != nil {
+			return map[string]string{
+				"org_name":            v.App.OrgName,
```
Comment (sfwn) [inline internal/tools/monitor/core/profile/profile.go:117]: Mixed case.
And you should use const.

File: internal/tools/monitor/oap/collector/core/pipeline/pipeline.go [hunk 5]
```diff
@@ -257,6 +258,16 @@ loop:
 					goto loop
 				}
 				data = tmp
+			case odata2.ProfileType:
+				tmp, err := pr.Processor.ProcessProfile(data.(*profile.ProfileIngest))
+				if err != nil {
+					p.Log.Errorf("Processor<%s> process data error: %s", pr.Name, err)
```
Comment (sfwn) [inline internal/tools/monitor/oap/collector/core/pipeline/pipeline.go:264]: process profile data error

File: internal/tools/monitor/oap/collector/plugins/exporters/clickhouse/builder/profile/profile.go [hunk 6]
```diff
@@ -0,0 +1,177 @@
...
+
+func NewBuilder(ctx servicehub.Context, logger logs.Logger, cfg *builder.BuilderConfig) (*Builder, error) {
+	bu := &Builder{
+		cfg:    cfg,
+		logger: logger,
+	}
+
+	ch, err := builder.GetClickHouseInf(ctx, odata.ProfileType)
+	if err != nil {
+		return nil, fmt.Errorf("get clickhouse interface: %w", err)
```
Comment (sfwn) [inline internal/tools/monitor/oap/collector/plugins/exporters/clickhouse/builder/profile/profile.go:67]: failed to get

File: internal/tools/monitor/oap/collector/plugins/processors/profile/provider.go [hunk 7]
```diff
@@ -12,108 +12,146 @@
...
+
+	appList := strings.Split(pi.Key.AppName(), ".")
+	app := &appmetadata.ApplicationMetadata{
+		FQName:          pi.Key.AppName(),
+		SpyName:         pi.SpyName,
+		SampleRate:      pi.SampleRate,
+		Units:           pi.Units,
+		AggregationType: pi.AggregationType,
+		SampleType:      appList[len(appList)-1],
+		OrgID:           pi.Key.Labels()["DICE_ORG_ID"],
```
Comment (sfwn) [inline internal/tools/monitor/oap/collector/plugins/processors/profile/provider.go:112]: ugly

**Review by sfwn — CHANGES_REQUESTED** [review]

File: cmd/monitor/streaming/bootstrap.yaml (hunk 1 above)
Comment (chengjoey) [reply cmd/monitor/streaming/bootstrap.yaml:693]: adjusted, WRITE_METRIC_TO_CLICKHOUSE_ENABLE -> WRITE_PROFILE_TO_CLICKHOUSE_ENABLE

File: cmd/monitor/streaming/bootstrap.yaml (hunk 2 above)
Comment (chengjoey) [reply cmd/monitor/streaming/bootstrap.yaml:695]: adjusted

File: cmd/monitor/streaming/bootstrap.yaml (hunk 3 above)
Comment (chengjoey) [reply cmd/monitor/streaming/bootstrap.yaml:698]: removed tenant_id_key

File: internal/tools/monitor/core/profile/profile.go (hunk 4 above)
Comment (chengjoey) [reply internal/tools/monitor/core/profile/profile.go:117]: done

File: internal/tools/monitor/oap/collector/core/pipeline/pipeline.go (hunk 5 above)
Comment (chengjoey) [reply internal/tools/monitor/oap/collector/core/pipeline/pipeline.go:264]: done

File: internal/tools/monitor/oap/collector/plugins/exporters/clickhouse/builder/profile/profile.go (hunk 6 above)
Comment (chengjoey) [reply internal/tools/monitor/oap/collector/plugins/exporters/clickhouse/builder/profile/profile.go:67]: done

**Review by sfwn — APPROVED** [review]

**Pull request merged**